### PR TITLE
Labeling configmaps as fmt

### PR DESCRIPTION
**Pull request by tomahawk28**
Toward https://github.com/giantswarm/giantswarm/issues/7763

We are not labeling on configmaps as fmt suggests. This PR is the first step to fix on it, later PR would backport this change.

### CODE REVIEW

File: service/controller/v26/resource/configmap/create.go
Comment (tomahawk28) [inline service/controller/v26/resource/configmap/create.go]: why it was `apiv1` is unclear, but set it up to `corev1` as we get used to it.

File: service/controller/v26/resource/configmap/desired.go
Comment (tomahawk28) [inline service/controller/v26/resource/configmap/desired.go]: Labeling as fmt suggests.

File: service/controller/v26/resource/configmap/update.go
Comment (tomahawk28) [inline service/controller/v26/resource/configmap/update.go]: the operator didn't compare labels and annotations before. Now we do it.

File: pkg/label/label.go
Comment (rossf7) [inline pkg/label/label.go]:
```suggestion
	// Cluster label is a new style label for ClusterID.
```

🇺🇸 nit

File: pkg/label/label.go
Comment (rossf7) [inline pkg/label/label.go]:
```suggestion
	// Organization label denotes tenant cluster's organization ID as displayed
```

Guest cluster is the old term.

File: service/controller/v26/resource/configmap/create.go
Comment (rossf7) [reply service/controller/v26/resource/configmap/create.go]: Like Tim said some of the code here is quite old. Thanks for improving it. <3

File: service/controller/v26/resource/configmap/create_test.go
Comment (rossf7) [inline service/controller/v26/resource/configmap/create_test.go]:
```suggestion
		resourceConfig.ProjectName = project.Name()
```

**Review by tfussell — APPROVED** [review]
Nice improvement!

File: service/controller/v26/cluster_resource_set.go
Comment (rossf7) [inline service/controller/v26/cluster_resource_set.go]: This is no longer needed. We should use `project.Name()`.

File: service/controller/v26/resource/configmap/desired.go
Comment (xh3b4sd) [inline service/controller/v26/resource/configmap/desired.go]: Are you sure we can simply drop these now? I cannot tell you what breaks now. We either need to understand this better or do a graceful migration. The latter could mean to keep the legacy labels, have a TODO issue and drop the legacy labels when `v26` is gone.

File: service/controller/v26/resource/configmap/update.go
Comment (rossf7) [reply service/controller/v26/resource/configmap/update.go]: We need to compare labels but didn't we have a problem comparing annotations in `app-operator`?

File: service/controller/v26/resource/configmap/resource.go
Comment (xh3b4sd) [inline service/controller/v26/resource/configmap/resource.go]: It is really cool that you do all this but when yo notice that there is more to it than one file you could think of providing a separate PR to make things easier.

File: service/controller/v26/resource/configmap/resource.go
Comment (xh3b4sd) [inline service/controller/v26/resource/configmap/resource.go]: This does also not look like labels changing.

File: service/controller/v26/resource/configmap/update_test.go
Comment (rossf7) [inline service/controller/v26/resource/configmap/update_test.go]:
```suggestion
		resourceConfig.ProjectName = project.Name()
```

File: service/controller/v26/resource/configmap/update.go
Comment (xh3b4sd) [reply service/controller/v26/resource/configmap/update.go]: Are you sure we should do this? I cannot tell you from the top of my head if we need to do this or maybe should not.

File: service/controller/v26/resource/configmap/desired.go
Comment (rossf7) [inline service/controller/v26/resource/configmap/desired.go]:
```suggestion
					label.ManagedBy:    project.Name(),
```

**Review by rossf7 — COMMENTED** [review]
This is nice but can be simpler by using `pkg/project`. Only doubt I have is on comparing annotations.

**Comment by rossf7** [comment]
Agree with Tim we need to be careful with labels and backwards compatibility. Also the fixing the package alias is nice but would be much easier to review in a separate PR.

File: service/controller/v26/resource/configmap/update.go
Comment (tomahawk28) [reply service/controller/v26/resource/configmap/update.go]: The only resource that using these configmaps is `deployment` resource in the later reconciliation step, and it used by objectName, so it's safe from this resource's point of view.
And I think we should make labels as same as kvm-operator desired.

File: service/controller/v26/resource/configmap/desired.go
Comment (tomahawk28) [reply service/controller/v26/resource/configmap/desired.go]: You're right, I think we can keep these labels at the moment and later drop it. I will address it afterward.

File: service/controller/v26/resource/configmap/resource.go
Comment (tomahawk28) [reply service/controller/v26/resource/configmap/resource.go]: To update labels as operator desired, we need to compare label values first between current and desired configmaps, so I need to this function to compare it, does it make sense to you, WDYT?

**Comment by tomahawk28** [comment]
I will revert import changes, it will come back after this PR.

File: service/controller/v26/resource/configmap/create_test.go
Comment (tomahawk28) [reply service/controller/v26/resource/configmap/create_test.go]: `config.ProjectName` isn't necessary by addressing your suggestion.

File: service/controller/v26/resource/configmap/update.go
Comment (tomahawk28) [reply service/controller/v26/resource/configmap/update.go]: Currently, we are comparing annotation even in app-operator https://github.com/giantswarm/app-operator/blob/703986b1568559b8fcaed5c0bc2d17d964ca2320/service/controller/app/v1/resource/configmap/resource.go#L80-L82
😱 Should we get rid of comparing these annotation values?

Comment (rossf7) [reply service/controller/v26/resource/configmap/update.go]: OK but in the chart resource we had problems which is what I was thinking of.

https://github.com/giantswarm/app-operator/blob/703986b1568559b8fcaed5c0bc2d17d964ca2320/service/controller/app/v1/resource/chart/resource.go#L89-L98

If we are not setting annotations in the desired state why do we need to compare them?

Comment (tomahawk28) [reply service/controller/v26/resource/configmap/update.go]: @rossf7 You're right, I deleted the part of comparing annotations for these configmaps. PTAL :)

**Comment by tomahawk28** [comment]
@tfussell Since so many changes have made after your approval, PTAL if you have a time :)

File: service/controller/v26/resource/configmap/desired.go
Comment (rossf7) [inline service/controller/v26/resource/configmap/desired.go]: Ordering is wrong.

**Review by rossf7 — APPROVED** [review]
LGTM

**Pull request merged**
